src/lib.rs

```diff
@@ -81,84 +81,153 @@ pub enum ParameterType {
+///
+///     ScaffoldDescription::new(opts)?.scaffold()
+/// # }
+///
+/// ```
+#[derive(Parser, Debug, Default)]
+#[command(author, version, about, long_about=None)]
 pub struct Opts {
     /// Specifiy your template location
-    #[structopt(name = "template", required = true)]
-    pub template_path: PathBuf,
+    #[arg(name = "template", required = true)]
+    template_path: PathBuf,
 
     /// Specifiy your template location in the repository if it's not located at the root of your repository
-    #[structopt(name = "repository_template_path", short = "r", long = "path")]
-    pub repository_template_path: Option<PathBuf>,
+    #[arg(name = "repository_template_path", short = 'r', long = "path")]
+    repository_template_path: Option<PathBuf>,
 
     /// Full commit hash, tag or branch from which the template is cloned
     /// (i.e.: "deed14dcbf17ba87f6659ea05755cf94cb1464ab" or "v0.5.0" or "main")
-    #[structopt(name = "git_ref", short = "t", long = "git_ref")]
-    pub git_ref: Option<String>,
+    #[arg(name = "git_ref", short = 't', long = "git_ref")]
+    git_ref: Option<String>,
 
     /// Specify the name of your generated project (and so skip the prompt asking for it)
-    #[structopt(name = "name", short = "n", long = "name")]
-    pub project_name: Option<String>,
+    #[arg(name = "name", short = 'n', long = "name")]
+    project_name: Option<String>,
 
     /// Specifiy the target directory
-    #[structopt(name = "target_directory", short = "d", long = "target_directory")]
-    pub target_dir: Option<PathBuf>,
+    #[arg(name = "target_directory", short = 'd', long = "target_directory")]
+    target_dir: Option<PathBuf>,
 
     /// Override target directory if it exists
-    #[structopt(short = "f", long = "force")]
-    pub force: bool,
+    #[arg(short = 'f', long = "force")]
+    force: bool,
 
     /// Append files in the target directory, create directory with the project name if it doesn't already exist but doesn't overwrite existing file (use force for that kind of usage)
-    #[structopt(short = "a", long = "append")]
-    pub append: bool,
+    #[arg(short = 'a', long = "append")]
+    append: bool,
 
     /// Ignored, kept for backwards compatibility [DEPRECATED]
-    #[structopt(short = "p", long = "passphrase")]
-    pub passphrase_needed: bool,
+    #[arg(short = 'p', long = "passphrase")]
+    passphrase_needed: bool,
 
     /// Specify if your private SSH key is located in another location than $HOME/.ssh/id_rsa
-    #[structopt(short = "k", long = "private_key_path")]
-    pub private_key_path: Option<PathBuf>,
+    #[arg(short = 'k', long = "private_key_path")]
+    private_key_path: Option<PathBuf>,
 
     /// Supply parameters via the command line in <name>=<value> format
-    #[structopt(long = "param")]
-    pub parameters: Vec<String>,
+    #[arg(long = "param")]
+    parameters: Vec<String>,
 }
 
-#[buildstructor::buildstructor]
 impl Opts {
```

Comment:
Can you also add `builder() -> Self` to try to keep retro compatibility please ?

Reply:
The `builder` in this case would just call `default()` right? TBH: the previous `new` API was kind of bad, the user has to know about all the internal details of the struct. One thought behind these new API is - user does not need to know about internal details (note: none of the `struct` fields are public). 

While I agree on the point that calling `default` is not obvious without reading the docs, but calling `builder` may be obvious because the user has previously used it, I guess we can simply make `builder` func which calls `default()` internally.